### PR TITLE
Delete system rules

### PR DESCRIPTION
**Pull request by urfolomeus**
This functionality allows us to delete system rules from Webfeeds. A system rule is one that is available to all feeds and not just an individual feed. Source rules work on all of a feed's items that come from an affected source, whereas template rules work on all of a feed's items regardless of their source.

In order to do this we lifted the ConfirmableForm component from the RAD project. This is in the process of being PRd into Dough, but we will refactor to that solution once it is available in a separate PR.

### CODE REVIEW

File: app/controllers/admin/source_rules_controller.rb [hunk 1]
```diff
@@ -28,6 +28,12 @@ def update
     end
   end
 
+  def destroy
+    rule = Rule.source_rules.find(params[:id])
+    rule.destroy
```
Comment (asmega) [inline app/controllers/admin/source_rules_controller.rb]: if you are not checking the result ie success/failure i'd call `.destroy!` instead

Comment (urfolomeus) [inline app/controllers/admin/source_rules_controller.rb]: Fair point, @asmega. Will update now.

**Comment by urfolomeus** [comment]
@asmega done

File: app/controllers/admin/source_rules_controller.rb (hunk 1 above)
Comment (urfolomeus) [inline app/controllers/admin/source_rules_controller.rb]: @asmega done.

**Comment by asmega** [comment]
:+1:

**Comment by urfolomeus** [comment]
We've now swapped out our local version of ConfirmableForm for the Dough one. This does not affect any other files.

**Comment by Tr4pSt3R** [comment]
Travis is failing. This seems to be a common failure. Probably an unmatched bracket somewhere


```ruby
1) widget previewing when headlines are excluded when the feed has one source item
     Failure/Error: Unable to find matching line from backtrace
     ActionView::Template::Error:
       couldn't find file 'dough/assets/js/lib/components/ConfirmableForm'
         (in /home/travis/build/moneyadviceservice/webfeeds/app/assets/javascripts/require_config.js.erb:7)
     # ./app/views/layouts/application.html.erb:51:in `_app_views_layouts_application_html_erb___4198187465288910535_72048520'
     # ./app/controllers/feeds_controller.rb:8:in `show'
```

**Comment by urfolomeus** [comment]
Thanks, @Tr4pSt3R. Caught and fixed. Was a typo on my part. :)

@asmega this shouldn't have changed anything that you reviewed other than to swap out our local implementation of ConfirmableForm, but you may want to give it a quick rescan?

**Comment by asmega** [comment]
:+1:

**Comment by Tr4pSt3R** [comment]
:100: :+1:

File: app/assets/stylesheets/components/_button.scss [hunk 2]
```diff
@@ -3,3 +3,23 @@
   padding: 14px 16px;
   line-height: 0.5;
 }
+
+.button--red {
```
Comment (neoeno) [inline app/assets/stylesheets/components/_button.scss]: I think the existing dough buttons are named semantically. So maybe `button--danger` would be better? @benbarnett?

Comment (benbarnett) [inline app/assets/stylesheets/components/_button.scss]: Yep concur

File: app/assets/stylesheets/components/_system_rule.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  }
+}
+
+.system-rule__actions {
+  @include column(12);
+  @include respond-to($mq-m) {
+    @include column(4);
+  }
+
+  text-align: right;
```
Comment (neoeno) [inline app/assets/stylesheets/components/_system_rule.scss]: I'd organise it this way:

```sass
.system-rule__actions {
  @include column(12);
  text-align: right;
  
  @include respond-to($mq-m) {
    @include column(4);
  }

  .button {
    float: right;
    margin-top: $baseline-unit;
    margin-left: $baseline-unit;
  }
}
```

That keeps the styles at each breakpoint together.

**Comment by mconnell** [comment]
@neoeno @benbarnett Thanks, I've amended the code with those suggestions :smiley:

**Comment by neoeno** [comment]
:+1:

**Comment by mconnell** [comment]
Thanks for the everyone!

**Pull request merged**